libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -623,52 +624,61 @@ static CK_RV prvCertAttParse( CK_ATTRIBUTE * pxAttribute,
         case ( CKA_VALUE ):
             *ppxCertificateValue = pxAttribute->pValue;
             *pxCertificateLength = pxAttribute->ulValueLen;
+            LogDebug( ( "Found CKA_VALUE attribute. Certificate value length was: %lu.", pxAttribute->ulValueLen ) );
             break;
 
         case ( CKA_LABEL ):
+            LogDebug( ( "Found CKA_LABEL attribute." ) );
 
             if( pxAttribute->ulValueLen <= pkcs11configMAX_LABEL_LENGTH )
             {
                 *ppxLabel = pxAttribute;
             }
             else
             {
-                LogError( ( "Label length was not in the valid range. Please check the value of pkcs11configMAX_LABEL_LENGTH" ) );
+                LogError( ( "Failed parsing certificate template. Label length"       \
+                            "was not in the valid range. Found %lu and expected %lu." \
+                            "Consider updating pkcs11configMAX_LABEL_LENGTH.", pxAttribute->ulValueLen, pkcs11configMAX_LABEL_LENGTH ) );
                 xResult = CKR_DATA_LEN_RANGE;
             }
 
             break;
 
         case ( CKA_CERTIFICATE_TYPE ):
+            LogDebug( ( "Found CKA_CERTIFICATE_TYPE." ) );
             ( void ) memcpy( pxCertificateType, pxAttribute->pValue, sizeof( CK_CERTIFICATE_TYPE ) );
 
             if( *pxCertificateType != CKC_X_509 )
             {
+                LogError( ( "Failed parsing certificate template. Certificate type was invalid. Expected CKC_X_509, but found %0X.", *pxCertificateType ) );
```

Comment:
Prefix hex values with `0x`?
```suggestion
                LogError( ( "Failed parsing certificate template. Certificate type was invalid. Expected CKC_X_509, but found 0x%0X.", *pxCertificateType ) );
```